### PR TITLE
[Routing] Add the `Requirement::UID_RFC9562` constant

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Issues        | -
| License       | MIT

Following https://github.com/symfony/symfony/pull/58329/files#r1768217660

### CODE REVIEW

**Review by stof — APPROVED** [review]

File: src/Symfony/Component/Routing/Tests/Requirement/RequirementTest.php [hunk 1]
```diff
@@ -224,10 +224,7 @@ public function testUidBase58KO(string $uid)
     }
 
     /**
-     * @testWith    ["00000000-0000-0000-0000-000000000000"]
-     *              ["ffffffff-ffff-ffff-ffff-ffffffffffff"]
-     *              ["01802c4e-c409-9f07-863c-f025ca7766a0"]
-     *              ["056654ca-0699-4e16-9895-e60afca090d7"]
+     * @dataProvider provideUidRfc4122
```
Comment (fabpot) [inline src/Symfony/Component/Routing/Tests/Requirement/RequirementTest.php:227]: Why changing this?

Comment (derrabus) [reply src/Symfony/Component/Routing/Tests/Requirement/RequirementTest.php:227]: I think it's because he wants to reuse these data sets for a second test.

**Review by xabbuh — APPROVED** [review]

File: src/Symfony/Component/Routing/Tests/Requirement/RequirementTest.php (hunk 1 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/Routing/Tests/Requirement/RequirementTest.php:227]: Yes, it's a way to explicitly use the dataset of RFC4122 with the new constant and show results should be the same

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @alexandre-daubois.

**Pull request merged**